src/excmds.ts

```diff
@@ -1763,17 +1763,24 @@ export async function tabclosealltoleft() {
 //#background
 export async function undo() {
     const current_win_id: number = (await browser.windows.getCurrent()).id
-    const sessions = await browser.sessions.getRecentlyClosed()
+    const sessions = await browser.sessions.getRecentlyClosed({ maxResults: 10 })
 
     // The first session object that's a window or a tab from this window. Or undefined if sessions is empty.
-    let closed = sessions.find(s => {
-        return "window" in s || (s.tab && s.tab.windowId == current_win_id)
+    const lastSession = sessions.find(s => {
+        if (s.window) {
+            return true
+        } else if (s.tab && s.tab.windowId) {
```

Comment:
Why did you move check down? If I close tabs in another window I don't want undo not to work...

Reply:
Oh right, oversight on my part. Got interrupted while rewriting that. The check was moved down in another implementation where it would have made sense.